week-1/fix/median.js

```diff
@@ -3,9 +3,12 @@
 // If you're in the week-1 directory, you can run npm test -- fix to run the tests in the fix directory
 
 function calculateMedian(list) {
+  const sortedList = [...list].sort((a,b)=> a-b);
   const middleIndex = Math.floor(list.length / 2);
-  const median = list.splice(middleIndex, 1)[0];
-  return median;
+  if(sortedList.length % 2 !== 0){
+    return sortedList[middleIndex];
+  }else{
+   return sortedList[middleIndex - 1] + sortedList[middleIndex] / 2;
```

Comment:
This line can return an element which is not in the array.
E.g. 3.5 for `calculateMedian([1, 3, 2, 4])`.
Median is the middle element of an array.
For the array above the result should be 2.
Can you explain in english first what is the intention in the line of code? Or using examples

Reply:
I was wrong saying that the result should be 2.
Shame on me. It's 2.5 really. You're right. 
Thanks for tolerating my asks to explain on english btw)
It's a very useful habbit

Reply:
Thank you so much for clarifying and your patience in explaining everything in detail. I really appreciate your guidance it’s always a learning moment for me. I may have been mistaken earlier, but your explanation helped me see things more clearly. You're an excellent teacher, and I’m grateful to have the opportunity to learn from you!